### PR TITLE
ECL-402 Other - Trust Entity Type

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/config/AppConfig.scala
Comment (TonyCOgilvie) [inline app/uk/gov/hmrc/economiccrimelevyregistration/config/AppConfig.scala:1]: I think you have left this in by mistake :)

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala [hunk 1]
```diff
@@ -18,22 +18,43 @@ package uk.gov.hmrc.economiccrimelevyregistration.navigation
...
 
 class CtUtrPageNavigator extends PageNavigator {
   override protected def navigateInNormalMode(registration: Registration): Call =
-    routes.CtUtrPostcodeController.onPageLoad(NormalMode)
+    registration.otherEntityJourneyData.ctUtr match {
+      case None    => routes.NotableErrorController.answersAreInvalid()
+      case Some(_) =>
+        registration.otherEntityJourneyData.entityType match {
+          case None        => routes.NotableErrorController.answersAreInvalid()
+          case Some(value) =>
```
Comment (TonyCOgilvie) [inline app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala:30]: I _think_ you can consolidate this logic into the case condition using something like case Some(value @ Trust) => and then Some(value) => will handle all other values that are not a Trust.

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala [hunk 2]
```diff
@@ -18,22 +18,43 @@ package uk.gov.hmrc.economiccrimelevyregistration.navigation
...
+    registration.otherEntityJourneyData.entityType match {
       case None        => routes.NotableErrorController.answersAreInvalid()
       case Some(value) =>
-        if (value) {
-          registration.otherEntityJourneyData.postcode match {
-            case None    => routes.CtUtrPostcodeController.onPageLoad(CheckMode)
-            case Some(_) => routes.OtherEntityCheckYourAnswersController.onPageLoad()
-          }
-        } else {
+        if (value == Trust) {
```
Comment (TonyCOgilvie) [inline app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala:43]: As per the previous comment.

**Review by TonyCOgilvie — COMMENTED** [review]
See individual comments

File: app/uk/gov/hmrc/economiccrimelevyregistration/config/AppConfig.scala
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/config/AppConfig.scala:1]: Oh yes, thank you :)

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala (hunk 1 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala:30]: Fixed this. Thank you for the suggestion, this is the first time that I've had a chance to learn about variable binding in pattern matching. I was always putting that away from some other time 😄

**Review by TonyCOgilvie — APPROVED** [review]

**Pull request merged**